### ISSUE TITLE
fix background poll and and finish it asap

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-ios#1157]
currently, `performFetchWithCompletionHandler` and `didReceiveRemoteNotification` let run the core IO for 10 seconds and terminate then.

it would be better to terminate once a fetch on the imap folders INBOX and DeltaChat is done (there is some info-message, but no dedicated event, maybe core needs some adaptions) _AND_ there are no more pending jobs (eg. sending messages out).

that way:
- we would usually return after a second or so and would get called faster again
- if a fetch really takes 10~20 seconds, that can be completed as well (after ~20 seconds, however, better terminate IO gracefully and call the completion handler properly) (absolute max. would be 30 seconds, however, at that point we risk to be brute-force-terminated and not to be called soon again)

**Comment by r10s** [comment]
some numbers:

on my iphone7, in the last 23½ h:
- 143 wakeups in total (68 remote, 75 local, in average one wakeup every 10 min)
- this should be 143 * 11 s (10 s wait, 1 s terminate) background time = **26 min**
- battery stats say 36 min in background, however, i assume, this includes the 30 s after leaving the app - so i have left the app 20 times (36 min - 26 min = 10 min -> 20 * 30 s) today, which is reasonable

so, assuming an fetch with no data is not much more than one second, the **26 min** can probably be reduced to a fragment - which, all in all, should get us a better reputation to be called more often (i do not expect to be called more than every 10 min, but there are phones with more apps where we are currently called less often)

**Comment by cyBerta** [comment]
This issue should be postponed until multi-account is implemented

**Comment by r10s** [comment]
just had a look, for the code, https://github.com/deltachat/deltachat-ios/pull/1221/files is still a pretty nice pattern.
